liquibase-standard/src/main/java/liquibase/changelog/StandardChangeLogHistoryService.java

```diff
@@ -249,18 +249,28 @@ public void init() throws DatabaseException {
             List<Map<String, ?>> md5sumRS = ChangelogJdbcMdcListener.query(getDatabase(), ex -> ex.queryForList(databaseChangeLogStatement));
 
             if (!md5sumRS.isEmpty()) {
-                String md5sum = md5sumRS.get(0).get("MD5SUM").toString();
-                if (!md5sum.startsWith(CheckSum.getCurrentVersion() + ":")) {
-                    executor.comment("DatabaseChangeLog checksums are an incompatible version.  Setting them to null " +
-                        "so they will be updated on next database update");
-                    databaseChecksumsCompatible = false;
-                    UpdateStatement updateStatement = new UpdateStatement(database.getLiquibaseCatalogName(),
-                            database.getLiquibaseSchemaName(), database.getDatabaseChangeLogTableName())
-                            .addNewColumnValue("MD5SUM", null);
-
-                    statementsToExecute.add(updateStatement);
-                }
+                //check if any checksum is not using the current version
+                databaseChecksumsCompatible = !md5sumRS.stream()
+                        .filter(m -> !m.get("MD5SUM").toString().startsWith(CheckSum.getCurrentVersion() + ":"))
+                        .findAny().isPresent();
```

Comment:
This stream statement could be simplified, I think:

```
                databaseChecksumsCompatible = md5sumRS.stream()
                        .filter(m -> !m.get("MD5SUM").toString().startsWith(CheckSum.getCurrentVersion() + ":"))
                        .noneMatch();
```

or this one, which is perhaps the most clear and easy to read:

```
                databaseChecksumsCompatible = md5sumRS.stream()
                        .filter(m -> m.get("MD5SUM").toString().startsWith(CheckSum.getCurrentVersion() + ":"))
                        .allMatch();
```